prusti-viper/src/encoder/spec_encoder.rs

```diff
@@ -375,6 +481,8 @@ impl<'p, 'v: 'p, 'tcx: 'v> SpecEncoder<'p, 'v, 'tcx> {
             curr_def_id = outer_def_id;
         }
 
+        let skip_first = self.encoder.encode_item_name(curr_def_id).contains("_closure_");
```

Comment:
This needs a comment. Why is this needed?

Reply:
I think this is to skip the closure's "self" argument. I am currently trying to figure out how the arguments get remapped, which should solve at least the "Local variable _1 not found." errors.

Reply:
I'm still not convinced about this. Maybe it was needed to avoid a bug that has been fixed in the meantime.

If before the replacement the (encoding of the) `self` argument is used in `curr_expr` then after the replacement it will still be there because of the `skip(1)`, but that would almost certainly be wrong because the remaining replacements could introduce a `_1` with a different type/meaning. So I guess that before the replacement `self` is not supposed to be in `curr_expr`. In that case, the `skip(1)` should have no effect, because the first replacement wouldn't match any place.

From what I understand we could remove `skip_first` because it's dead code. However, I cannot find an usage of `encode_assertion` which actually passes the encoding of the `self` argument that should be skipped by `skip(1)`. In my view the implementation of `encode_{pre|post}_spec_func` should prepend `encoded_args` with a dummy variable named like `_nonexistent_`, but that does not happen.

(As a side note, I would prefer having closure-specific logic on the caller side of `encode_assertion`, where it should be clear whether we are encoding a closure, rather than inside `encode_assertion` where we can only use the heuristic on the encoded item name.)

Reply:
> [...] I would prefer having closure-specific logic on the caller side of `encode_assertion`, where it should be clear whether we are encoding a closure [...]

I agree. `skip_first` will eventually not be needed, once we figure out how to properly encode the closure's self type with snapshots.